# denoise
image_denoising:
  hist_equalization: true
  denoise: false
  h: 10.0 
  templateWindowSize: 7
  searchWindowSize: 21
# estimator
pose_estimation:
  pattern: 0
  solvePnPMethod: 7 
# aruco detector
aruco_detection:
  adaptiveThreshWinSizeMin: 3
  adaptiveThreshWinSizeMax: 23
  adaptiveThreshWinSizeStep: 10
  adaptiveThreshConstant: 7.0
  minMarkerPerimeterRate: 0.03
  maxMarkerPerimeterRate: 4.0
  polygonalApproxAccuracyRate: 0.03
  minCornerDistanceRate: 0.05
  minDistanceToBorder: 3
  minMarkerDistanceRate: 0.125
  minGroupDistance: 0.21
  cornerRefinementMethod: 2
  cornerRefinementWinSize: 5
  relativeCornerRefinmentWinSize: 0.3
  cornerRefinementMaxIterations: 30
  cornerRefinementMinAccuracy: 0.1
  markerBorderBits: 1
  perspectiveRemovePixelPerCell: 4
  perspectiveRemoveIgnoredMarginPerCell: 0.13
  maxErroneousBitsInBorderRate: 0.35
  minOtsuStdDev: 5.0
  errorCorrectionRate: 0.6
  aprilTagQuadDecimate: 0.0
  aprilTagQuadSigma: 0.0
  aprilTagMinClusterPixels: 5
  aprilTagMaxNmaxima: 10
  aprilTagCriticalRad: 0.1745329201221466
  aprilTagMaxLineFitMse: 10.0
  aprilTagMinWhiteBlackDiff: 5
  aprilTagDeglitch: 0
  detectInvertedMarker: false
  useAruco3Detection: false
  minSideLengthCanonicalImg: 32
  minMarkerLengthRatioOriginalImg: 0.0
# apriltag detector
apriltag_detection:
  nthreads: 4
  quad_decimate: 0.0
  quad_sigma: 0.0
  refine_edges: true
  decode_sharpening: 0.5
  decision_margin: 20 
  max_hamming: 10
  min_marker_width : 0 
  min_marker_height: 0
